### DISCUSSION TITLE
Serverless?

### DISCUSSION THREAD

**Discussion by PedroHase** [geldata/gel#3836] [General]
Hey all, I just stumbled upon EdgeDB and I really like the idea of it! However, I do not see any real advantage yet over using "classic" ORMs like Prisma if we need to deploy a full-fledged server instance just for the "database" (i.e. the database management instance) when we could just use Prisma that almost does the same thing sans the need for an additional server (i.e. defining the data models, querying, Typescript types, etc).

But I read on the website about the "asynchronous core" and "**serverless** deployments". Therefore my question, will it be possible in the future to deploy EdgeDB on services like Lambda or Cloud Functions?

**Comment by elprans** [comment]
EdgeDB is different from a "classic ORM" in a number of ways:

- it is language- and framework- independent, whereas a classic ORM, like Prisma is language-specific, making it hard to switch languages or have a multi-language project and keep all the benefits;
- EdgeDB is not a simple ORM -- it's essentially an alternative frontend to PostgreSQL storage and query engine. This is important, because we can implement advanced query and schema features using PostgreSQL-only capabilities, and with [great performance](https://www.edgedb.com/blog/why-orms-are-slow-and-getting-slower#javascript-orms-full-report);
- EdgeQL is a [better SQL](https://www.edgedb.com/blog/we-can-do-better-than-sql), whereas most ORMs only have a fairly limited set of query features;
- Classic ORMs usually connect directly to the database server, and most traditional relational databases don't scale very well in the number of concurrent connections, thus you might start having issues in a true distributed/serverless environment where you have a bunch of lambdas all connecting to the same server.  EdgeDB has a builtin connection pool, so you don't have to worry about that and have thousands of simultaneous connections to an EdgeDB server.  Additionally, our protocol and client libraries are designed to be cloud-first, i.e. have high tolerance to network errors and concurrent transaction contention.
- Lastly, to answer the subject in question, we are working on EdgeDB-as-a-service (private beta coming soon), so you don't have to deal with server deployment unless you want to.

**Reply by riolly** [reply]
EdgeDB compelled me to switch from prisma. However, [planetscale infrastructure](https://principles.planetscale.com/) which _serverless by default_ and _linearly horizontally scalable_ makes me nervous not using it.

I hope EdgeDB can bring those values to EdgeDB Cloud.
And I know your that team is aware of that.
